code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java

```diff
@@ -257,6 +263,16 @@ public void show(final boolean withMessagingDelegateControl) {
     /** Dismisses the message. */
     @Override
     public void dismiss() {
+        if (messagesMonitor.dismiss()) {
+            return;
+        }
+
+        // notify listeners
+        listener.onDismiss(this);
```

Comment:
Can you remove all these changes and keep the old logic? You should attempt to remove a message only if it has been shown earlier?

Reply:
i have reverted this to how it was previously